stellargraph/core/graph_networkx.py

```diff
@@ -772,6 +772,29 @@ def to_adjacency_matrix(self, nodes: Optional[Iterable] = None):
             self._graph, dtype="float32", weight=self._edge_weight_label, format="coo"
         )
 
+    def to_networkx(self):
+        # This is implemented generically to not be tied to the (outgoing) networkx format.
+        if self.is_directed():
+            graph = nx.MultiDiGraph()
+        else:
+            graph = nx.MultiGraph()
+
+        types = self.node_types
+
+        for ty in types:
+            node_ids = self.nodes_of_type(ty)
+            features = self.node_features(node_ids, node_type=ty)
+
+            for node_id, node_features in zip(node_ids, features):
+                graph.add_node(
+                    node_id,
+                    **{self._feature_attr: node_features, self._node_type_attr: ty},
```

Comment:
With #598 it seems like if `self._feature_attr` and `self._node_type_attr` are promoted, we can also promote this implementation - thoughts?

Reply:
Good point, but I'm not sure I want to promote those attributes at the moment, they're a bit peculiar and maybe don't fit into a Pandas/Numpy world very well.